mop/brokers/asassn.py

```diff
@@ -0,0 +1,224 @@
+import os
+from django import forms
+import requests
+import urllib
+from urllib.error import HTTPError
+from urllib.request import Request, urlopen
+from astropy.time import Time, TimezoneInfo
+import datetime
+import pandas as pd
+import lxml.html as lh
+import mechanicalsoup
+from bs4 import BeautifulSoup
+from astropy.coordinates import SkyCoord
+import astropy.units as unit
+import json
+from html_table_parser.parser import HTMLTableParser
+
+
+from tom_targets.models import Target
+from tom_alerts.alerts import GenericBroker, GenericQueryForm
+from tom_dataproducts.models import ReducedDatum
+
+BROKER_URL = 'http://www.astronomy.ohio-state.edu/asassn/transients.html' #should be a known variable?
+photometry = 'https://asas-sn.osu.edu/photometry'
+
+
+class ASASSNBroker():
+    def __init__(self, name):  
+        self.name = name 
+    #name='ASAS-SN'
+
+    def open_webpage(self):
+        page_status_code=requests.get(BROKER_URL).status_code
+        return page_status_code
+
+    def retrieve_transient_table(self):
+        page=requests.get(BROKER_URL)
+        doc = lh.fromstring(page.content)
+        tr_elements = doc.xpath('//tr')
+        col=[]
+        i=0
+        for t in tr_elements[0]:
+            i+=1
+            content=t.text_content()
+            col.append((content,[]))
+        for j in range(1,len(tr_elements)):
+            #T is our j'th row
+            T=tr_elements[j]
+            #If row is not of size 12, the //tr data is not from our table 
+            if len(T)!=12:
+                break
+            i=0
+            #Iterate through each element of the row
+            for t in T.iterchildren():
+                data=t.text_content() 
+                #Check if row is empty
+                if i>0:
+                #Convert any numerical value to integers
+                    try:
+                        data=int(data)
+                    except:
+                        pass
+                #Append the data to the empty list of the i'th column
+                col[i][1].append(data)
+                #Increment i for the next column
+                i+=1
+        return col
+
+    def retrieve_microlensing_coordinates(self):
+        transienttable = self.retrieve_transient_table()
+        listofindices=[]
+        i=0
+        newlist = [str(s) for s in transienttable[11][1]]   
+        for m in newlist:
+            if 'microlensing' in m or 'Microlensing' in m:
+                listofindices.append(i)
+            i+=1
+        fullralist = []
```

Comment:
It's not clear to me that you need to initialize the fullralist etc to empty lists here, since you immediately set them from an implicit loop below.  Are these four lines legacy code?

Reply:
You're right, I initialized them before figuring out how to populate them. Previously, I was trying to use .append() to populate, so they needed to be initialized